packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1057,6 +1057,24 @@ class SliverFillViewport extends SliverMultiBoxAdaptorWidget {
   void updateRenderObject(BuildContext context, RenderSliverFillViewport renderObject) {
     renderObject.viewportFraction = viewportFraction;
   }
+
+  @override
+  double estimateMaxScrollOffset(
+    SliverConstraints constraints,
+    int firstIndex,
+    int lastIndex,
+    double leadingScrollOffset,
+    double trailingScrollOffset,
+  ) {
+    final double padding = (1 - viewportFraction) * constraints.viewportMainAxisExtent;
+    if (delegate.estimatedChildCount == null) {
+      return trailingScrollOffset + padding / 2;
```

Comment:
Why is this a good approximation?

Reply:
Without `delegate.estimatedChildCount` it's hard to estimate the max scroll offset. Should I just return `double.infinity` or `null`?

Reply:
Changed to `double.infinity`